### PR TITLE
Cranelift CLIF Fuzzer add jump tables and `br_table`

### PR DESCRIPTION
**Pull request by afonso360**
Hey!

A small addition to #3094. This is the last planned branch instruction for the CLIF fuzzer (since [it looks like `brif`/`brff` are going to be removed](https://github.com/bytecodealliance/wasmtime/issues/3249)), and means that we have completed the branch / jump part of the fuzzer 🎉 

This change independently found #3100 which was fixed in #3282.

### CODE REVIEW

File: cranelift/fuzzgen/src/function_generator.rs [hunk 1]
```diff
@@ -229,6 +237,16 @@ where
...
     }
 
+    /// Valid blocks for jump tables have to have no parameters in the signature, and must also
+    /// not be the first block.
+    fn generate_valid_jumptable_target_blocks(&mut self) -> Vec<Block> {
+        self.blocks[1..]
+            .iter()
+            .filter(|(_, sig)| sig.len() == 0)
+            .map(|(b, _)| *b)
+            .collect()
```
Comment (cfallin) [inline cranelift/fuzzgen/src/function_generator.rs:247]: The "empty blockparams" condition is of course necessary, but how do we guard against cases where no blocks fit this requirement, and we end up doing a `choose()` on an empty list (which will panic)? Possibly I'm just missing this invariant elsewhere -- if so, could you add a comment describing why it'll always succeed, and maybe an `assert!(target_blocks.len() > 0)` or somesuch?

**Review by cfallin — APPROVED** [review]
LGTM overall!

Question below about how we ensure we always have at least one block that can go into a jumptable; with explanation in comment, I'm happy to merge.

File: cranelift/fuzzgen/src/function_generator.rs (hunk 1 above)
Comment (afonso360) [reply cranelift/fuzzgen/src/function_generator.rs:247]: This already happens all the time. i.e. we are never guaranteed to generate variables of any type, so any instruction that requests a value of a type that doesn't exist ends up doing a choose on an empty block.

https://github.com/bytecodealliance/wasmtime/blob/3db8043a0284dd40e2859f8df3e05e730a706624/cranelift/fuzzgen/src/function_generator.rs#L194-L198

What ends up happening here, is that choose returns an error, we bubble the error up and into cargo-fuzz, which marks this input as invalid and starts another round of fuzzing.

Comment (cfallin) [reply cranelift/fuzzgen/src/function_generator.rs:247]: Ah, great, the key bit is that the failure bubbles up; I couldn't recall if `choose()` just panics in this case instead. All good then -- I'll go ahead and merge, thanks!

**Pull request merged**